LFS-497: MaterialTable cell fonts are inconsistent with the rest of the theme

OrdiNeu opened the pull request:
[Jira link](https://phenotips.atlassian.net/browse/LFS-497)

The font on selecting a SubjectType or a Subject should now be using the same sans serif font as the rest of the theme.

OrdiNeu committed:
LFS-497: MaterialTable cell fonts are inconsistent with the rest of the theme

sdumitriu reviewed (commented):
This seems fishy. Are you sure there's no better way?

sashaandjic commented:
Looks good.
![image](https://user-images.githubusercontent.com/23243244/88258392-15b7a280-cc8e-11ea-8267-c88db51e45b3.png)

OrdiNeu commented:
@sdumitriu 
> This seems fishy. Are you sure there's no better way? 

As far as I can tell, it's the only way to get it to obey the Material-UI-Cell style, since they [overwrite these three styles with "inherit"](https://github.com/mbrn/material-table/blob/master/src/components/m-table-cell.js#L118), which prevents the class style applied to them via Material-UI from working.

sdumitriu commented:
I see, my suggestion was to use `inherit` if the Material Table widgets do force it to something else, but it seems that inheriting is the real cause. Which means that the issue is that we don't have a default font family set in our styles, so the inherited value is the browser default. Which makes me think that the font family is explicitly set for EVERY ELEMENT on the page... This is bad theming, we really need to clean up our theme soon.

I'll leave the choice to you: define a global font family on the `<body>` and delete the custom styles, or just keep the current quickfix and deal with this later.

OrdiNeu commented:
> Which makes me think that the font family is explicitly set for EVERY ELEMENT on the page... This is bad theming, we really need to clean up our theme soon. 

As far as I can see, it's being set through classes such as `.MuiTableCell-root` on every element, and there's no top level element with the font specified.

OrdiNeu committed:
LFS-497: MaterialTable cell fonts are inconsistent with the rest of the
theme
Remove the useage of defaultFonts from our themeStyle, instead
specifying them in the <body> tag

sdumitriu commented:
@OrdiNeu are you still working on this, or should we just merge the quickfix as it is?

OrdiNeu commented:
> @OrdiNeu are you still working on this, or should we just merge the quickfix as it is?
The font should be applied globally now via the `<body>`

@sashaandjic I think this needs a once-more-over due to the recent overhaul.

sdumitriu reviewed (approved):
I like the cleanup this provides, but I am a bit worried about styling the HTML directly. I think this will have to do at the moment, but in a later app reorganization I think we'll have a global application component that will have this style applied globally.

sashaandjic reviewed (approved):
re-tested
